### PR TITLE
add Socket.prototype & Readable.prototype

### PR DESCRIPTION
**Pull request by Geczy**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [ ] Code changes

### CODE REVIEW

File: src/js/node/stream.js [hunk 1]
```diff
@@ -2282,9 +2282,13 @@ var require_readable = __commonJS({
         }
       });
     }
-    Readable.prototype = {};
-    ObjectSetPrototypeOf(Readable.prototype, Stream.prototype);
-    ObjectSetPrototypeOf(Readable, Stream);
+    // Set Readable's prototype to Stream's prototype,
+    // so instances of Readable will inherit from Stream
+    Readable.prototype = Stream.prototype;
```
Comment (paperclover) [inline src/js/node/stream.js:2287]: this isn't correct. i'll fix this on my pr

Comment (Geczy) [reply src/js/node/stream.js:2287]: yes please, when is it available?

Comment (Geczy) [reply src/js/node/stream.js:2287]: @paperdave any update pls

Comment (Geczy) [reply src/js/node/stream.js:2287]: pls @paperdave

Comment (Geczy) [reply src/js/node/stream.js:2287]: @paperdave

**Comment by Jarred-Sumner** [comment]
Sorry, this PR no longer applies anymore. The intent of it has since been fixed.

**Comment by Geczy** [comment]
@Jarred-Sumner it doesn't look like it's super fixed, node:stream still doesn't exist? 

```js
2024-05-19 09:56:19 TypeError: undefined is not an object
2024-05-19 09:56:19       at node:stream:1738:207
2024-05-19 09:56:19       at new Connection (/app/node_modules/steam-fork/lib/connection.js:10:49)
2024-05-19 09:56:19       at /app/node_modules/steam-fork/lib/steam_client.js:70:15
2024-05-19 09:56:19       at new Dota (/app/packages/steam/src/steam.ts:351:34)
2024-05-19 09:56:19       at getInstance (/app/packages/steam/src/steam.ts:551:16)
2024-05-19 09:56:19       at module code (/app/packages/steam/src/index.ts:15:2)
```

**Comment by Geczy** [comment]
here's the contents of connection.js

```
module.exports = Connection;

var util = require('util');
var Socket = require('net').Socket;
var MAGIC = 'VT01';

util.inherits(Connection, Socket);

function Connection() {
  Socket.call(this);
  this.on('readable', this._readPacket.bind(this));
}
```

i'm using bun 1.1.8

**Pull request closed without merging**